File: verityx/src/main/java/com/example/verityx/service/SysInfoService.java
package com.example.verityx.service;

import com.baomidou.mybatisplus.core.metadata.IPage;
import com.example.verityx.dto.SysVersionInfoDTO;
import com.example.verityx.dto.SysLicenseDTO;
import com.example.verityx.dto.SysUpgradeRecordDTO;
import com.example.verityx.dto.SysAboutDTO;
import com.example.verityx.dto.request.SysVersionInfoRequest;
import com.example.verityx.dto.request.SysLicenseRequest;
import com.example.verityx.dto.request.SysUpgradeRequest;
import com.example.verityx.dto.request.SysAboutRequest;

import java.util.List;

/**
 * 系统信息Service接口
 */
public interface SysInfoService {
    
    // 版本信息相关方法
    
    /**
     * 获取当前版本信息
     * 
     * @return 当前版本信息
     */
    SysVersionInfoDTO getCurrentVersion();
    
    /**
     * 获取版本历史记录
     * 
     * @return 版本历史记录列表
     */
    List<SysVersionInfoDTO> getVersionHistory();
    
    /**
     * 添加新版本信息
     * 
     * @param request 版本信息请求
     * @return 新增的版本信息ID
     */
    Long addVersionInfo(SysVersionInfoRequest request);
    
    /**
     * 更新版本信息
     * 
     * @param request 版本信息请求
     * @return 是否更新成功
     */
    boolean updateVersionInfo(SysVersionInfoRequest request);
    
    /**
     * 设置当前版本
     * 
     * @param id 版本ID
     * @return 是否设置成功
     */
    boolean setCurrentVersion(Long id);
    
    // 许可证信息相关方法
    
    /**
     * 获取当前许可证信息
     * 
     * @return 当前许可证信息
     */
    SysLicenseDTO getCurrentLicense();
    
    /**
     * 获取许可证列表
     * 
     * @return 许可证列表
     */
    List<SysLicenseDTO> getLicenseList();
    
    /**
     * 添加许可证信息
     * 
     * @param request 许可证信息请求
     * @return 新增的许可证信息ID
     */
    Long addLicense(SysLicenseRequest request);
    
    /**
     * 更新许可证信息
     * 
     * @param request 许可证信息请求
     * @return 是否更新成功
     */
    boolean updateLicense(SysLicenseRequest request);
    
    /**
     * 验证许可证密钥
     * 
     * @param licenseKey 许可证密钥
     * @return 验证结果
     */
    boolean validateLicenseKey(String licenseKey);
    
    // 系统升级相关方法
    
    /**
     * 获取升级记录列表（分页）
     * 
     * @param page 页码
     * @param size 每页大小
     * @param status 状态
     * @return 分页结果
     */
    IPage<SysUpgradeRecordDTO> getUpgradeRecordPage(int page, int size, String status);
    
    /**
     * 获取升级记录详情
     * 
     * @param id 升级记录ID
     * @return 升级记录详情
     */
    SysUpgradeRecordDTO getUpgradeRecordDetail(Long id);
    
    /**
     * 执行系统升级
     * 
     * @param request 升级请求
     * @return 升级记录ID
     */
    Long executeUpgrade(SysUpgradeRequest request);
    
    /**
     * 获取最近一次升级记录
     * 
     * @return 最近一次升级记录
     */
    SysUpgradeRecordDTO getLatestUpgradeRecord();
    
    // 关于系统相关方法
    
    /**
     * 获取系统基本信息
     * 
     * @return 系统基本信息
     */
    SysAboutDTO getSystemInfo();
    
    /**
     * 更新系统基本信息
     * 
     * @param request 系统基本信息请求
     * @return 是否更新成功
     */
    boolean updateSystemInfo(SysAboutRequest request);
    
    /**
     * 获取服务条款
     * 
     * @return 服务条款
     */
    String getTermsOfService();
    
    /**
     * 获取隐私政策
     * 
     * @return 隐私政策
     */
    String getPrivacyPolicy();
}

File: front/src/api/sysInfoService.js
import { http } from '@/utils/request'

/**
 * 系统信息相关API
 */
export const SysInfoApi = {
  /**
   * 获取系统基本信息
   * @returns {Promise}
   */
  getSystemInfo() {
    return http.get('/api/system/info/about')
  },

  /**
   * 更新系统信息
   * @param {Object} data 系统信息数据
   * @returns {Promise}
   */
  updateSystemInfo(data) {
    return http.put('/api/system/info/about', data)
  },

  /**
   * 获取服务条款
   * @returns {Promise}
   */
  getTermsOfService() {
    return http.get('/api/system/info/about/terms')
  },

  /**
   * 获取隐私政策
   * @returns {Promise}
   */
  getPrivacyPolicy() {
    return http.get('/api/system/info/about/privacy')
  },

  /**
   * 获取当前版本信息
   * @returns {Promise}
   */
  getCurrentVersion() {
    return http.get('/api/system/info/version/current')
  },

  /**
   * 获取版本历史
   * @returns {Promise}
   */
  getVersionHistory() {
    return http.get('/api/system/info/version/history')
  },

  /**
   * 添加版本信息
   * @param {Object} data 版本信息数据
   * @returns {Promise}
   */
  addVersionInfo(data) {
    return http.post('/api/system/info/version', data)
  },

  /**
   * 更新版本信息
   * @param {Object} data 版本信息数据
   * @returns {Promise}
   */
  updateVersionInfo(data) {
    return http.put('/api/system/info/version', data)
  },

  /**
   * 设置当前版本
   * @param {string|number} id 版本ID
   * @returns {Promise}
   */
  setCurrentVersion(id) {
    return http.put(`/api/system/info/version/${id}/current`)
  },

  /**
   * 获取当前许可证信息
   * @returns {Promise}
   */
  getCurrentLicense() {
    return http.get('/api/system/info/license/current')
  },

  /**
   * 获取许可证列表
   * @returns {Promise}
   */
  getLicenseList() {
    return http.get('/api/system/info/license/list')
  },

  /**
   * 添加许可证
   * @param {Object} data 许可证数据
   * @returns {Promise}
   */
  addLicense(data) {
    return http.post('/api/system/info/license', data)
  },

  /**
   * 更新许可证信息
   * @param {Object} data 许可证数据
   * @returns {Promise}
   */
  updateLicense(data) {
    return http.put('/api/system/info/license', data)
  },

  /**
   * 验证许可证密钥
   * @param {string} licenseKey 许可证密钥
   * @returns {Promise}
   */
  validateLicenseKey(licenseKey) {
    return http.post('/api/system/info/license/validate', { licenseKey })
  },

  /**
   * 获取最近的升级记录
   * @returns {Promise}
   */
  getLatestUpgradeRecord() {
    return http.get('/api/system/info/upgrade/latest')
  },

  /**
   * 获取升级记录列表
   * @param {number} page 页码
   * @param {number} pageSize 每页大小
   * @param {string} status 过滤状态
   * @returns {Promise}
   */
  getUpgradeRecordList(page = 1, pageSize = 10, status = '') {
    return http.get('/api/system/info/upgrade/list', { 
      params: { page, pageSize, status }
    })
  },

  /**
   * 获取升级记录详情
   * @param {string|number} id 升级记录ID
   * @returns {Promise}
   */
  getUpgradeRecordDetail(id) {
    return http.get(`/api/system/info/upgrade/${id}`)
  },

  /**
   * 执行系统升级
   * @param {Object} data 升级配置数据
   * @returns {Promise}
   */
  executeUpgrade(data) {
    return http.post('/api/system/info/upgrade', data)
  }
}

export default SysInfoApi; 